app/models/article.rb

```diff
@@ -813,25 +813,34 @@ def parse_date(date)
     published_at || date || Time.current
   end
 
+  # When an article is saved, it ensures that the tags that were adjusted by moderators and admins
+  # remain adjusted. We do not allow the author to add or remove tags that were previously added or
+  # removed by moderators and admins.
+  #
+  # This method is called before validation, so that the tag_list can be validated.
+  #
+  # @return [String] an array of tag names.
   def validate_tag
-    # remove adjusted tags
-    remove_tag_adjustments_from_tag_list
-    add_tag_adjustments_to_tag_list
+    distinct_tag_adjustments = TagAdjustment.where(article_id: id, status: "committed")
```

Comment:
We had to add this query because previously, we could only ever remove or add an adjustment for a tag on an article once, because once a tag moderator made an adjustment than no other moderator could make an adjustment. This is why we added a unique index on the TagAdjustment model for an article_id_and_tag_name. Hence, when we ran `validate_tag` on an article model to show the final tag lists (https://github.com/forem/forem/blob/ed853f7e1ec82fe90997d624857ba1f465573927/app/models/article.rb#L810) we previously only ever had the tag appearing once in either the “tags_to_add” bucket OR “tags_to_remove” bucket as per this code. 

Now, we need to get the latest adjustment for a unique tag_id.